ext/http/http_next.rs

```diff
@@ -186,20 +212,23 @@ pub fn op_http_upgrade_raw(
 #[smi]
 pub async fn op_http_upgrade_websocket_next(
   state: Rc<RefCell<OpState>>,
-  #[smi] slab_id: SlabId,
+  external: *const c_void,
   #[serde] headers: Vec<(ByteString, ByteString)>,
 ) -> Result<ResourceId, AnyError> {
-  let mut http = slab_get(slab_id);
+  let http =
```

Comment:
Nothing is ensuring the external that is cast here (and elsewhere in this code) is actually of the shape we are expecting? Ie that it's really an `Rc<T>` and not a an `Rc<T2>` or `Arc<T>`.

Users can arbitrarily call ops, with arbitrary arguments (anything exposed to JS, internal or not, is untrusted). As such, we need to verify the argument here is of the correct type, because a user could obtain a different `v8::External` from a different op, pass it in here, where we cast it to an expected type and then read / write to that memory, which results in memory corruption / panic / segfault at best, but arbitrary code exec at worst.

Please clarify if I am misunderstanding the safety implications here, but I think as it stands right now, this is not landable.

Reply:
This code is now using the new type-checking ExternalPointer from deno_core to ensure that all external references are safe.